src/LuaTransformer.ts

```diff
@@ -4995,57 +5006,48 @@ export class LuaTransformer {
                         expression
                     );
                 }
-                if (!isWithinExpressionStatement) {
+                if (expression.parent.kind !== ts.SyntaxKind.ExpressionStatement) {
```

Comment:
That won't work if it's gone through `transformExpressionStatement` as an expression

Reply:
Seems like that would cause a bug but I can't seem to create some code to break this. What kind of code would do you think would break this?

Reply:
Just like above, anything that calls `transformExpressionStatement` directly with an expression

Reply:
It looks it is working as expected. It doesn't break when it goes through `transformExpressionStatement` as an Expression and will throw the Exception appropriately

```ts
/** @LuaTable */
declare class LuaTable<K extends {} = {}, V = any> {
  readonly length: number;
  set(key: K, value: V): number;
  get(key: K): V;
}
let y;
let t = new LuaTable();
for (let x = 0; x < 10; t.set("x", 4)) {} // ❌
for (let x = 0; x < 10; x = t.set("x", 4)) {} // ❌
let x = 4 + (y = t.set('', 2)); // ❌
let xx = 4 + t.set('', 2); // ❌
let xxx = t.set('', 2); // ❌
let y = 4 + (y = t.get('')); // ✔
let yy = 4 + t.get(''); // ✔
let yyy = t.get(''); // ✔
```

`transformExpressionStatement` is only called with an ExpressionStatement from `transformStatement` and this would be the only case where `luaTable.set` would be allowed

Reply:
It shouldn't throw in these cases. The whole point of passing an expression to `transformExpressionStatement` is to assume that this expression is used in statement context, even if it's not a part of ExpressionStatement.